consensus_pbft.go

```diff
@@ -435,11 +433,6 @@ func (p *Pbft) runValidateState(ctx context.Context) { // start new round
 			// change to commit state just to get out of the loop
 			p.setState(CommitState)
 		}
-
-		// set the attributes of this span once it is done
-		p.setStateSpanAttributes(span)
```

Comment:
This might be the reason why there was an inter-span. Imagine that the system gets locked in one of the cycles, we would not get any traces at all because it is waiting to exit.

Reply:
Are you referring that `runXXXState` (`runValidateState`, `runRoundChangeState` etc.) execution could stuck? How is that even possible (at least with the current implementation)? Eventually a timeout would elapse and the execution of a  PBFT state handler would come to an end.
If PBFT consensus is stuck (or locked as you are referring) it still means that we would have transitions as such: `AcceptState` -> `RoundChangeState` -> `AcceptState` etc.

So do you still find useful leaving inter-spans or not?

Reply:
Can you elaborate on this?

Reply:
> Can you elaborate on this?

My point was to keep only single span through entire PBFT state handler (instead of reporting spans every time a message is read from the queue). 
Both approaches have positive and negative aspects: 
- **creating a span for each message retrieval** (like it was before) 
   pros: we are sure that we wouldn't miss any information, if (for some strange reason), we get stuck any of the `runXXXState` function.
   cons: in order to track which messages are aggregated through given PBFT state, we would need to analyze 
   separately each span, which can be tedious, since we don't have reported all the accumulated messages at the single 
   place, but reporting it one by one.
- **creating a single span for entire PBFT state handler**
  pros: all the information are aggregated and easily consumed
  cons: Ferran's concern [here](https://github.com/0xPolygon/pbft-consensus/pull/76#discussion_r940043739). Although I   don't see that it is likely to happen (as we have timeouts as a mechanism to break execution if it lasts too long).
  
On the end, I'm good with both approaches, just need to make an agreement and I'll accommodate it.

Reply:
Yes I like this approach @Stefan-Ethernal, I was asking @ferranbt to explain his concerns 😄 because as you said, I don't see any stopper

Reply:
As discussed, lets change this except for RoundChange state.